[Documentation] 01 -  Introduction

davidblader opened the pull request.

driusan commented on docs/01-WhatIsLORISMRI.md:
```diff
@@ -1 +1,12 @@
 # 1.0 - What is LORIS-MRI?
+LORIS-MRI is the backbone of the imaging component that makes up LORIS. These documents assume you have some knowledge regarding LORIS and a functioning installation. For information regarding LORIS itself, please consult the [LORIS wiki][1].
```
Can you fix the length of these lines? It's hard to read with everything on one line.

(the easiest way I know of to consistently fix it for paragraphs is to pipe the file through the unix `fmt` command..

christinerogers commented on docs/01-Introduction.md:
```diff
@@ -1,5 +1,55 @@
+regarding LORIS itself, please consult the [LORIS wiki][1].
 
-## 1.2 How does LORIS work?
+LORIS-MRI is responsible for the processing, visualizing, and archiving
```
This is seriously great.  
I would suggest specifying _what_ it is: 
- a separate repo, so that it can be installed on the file server not the web server
- a set of libraries and scripts that handle imaging file formats, pre-processing, quality checks and insertion. (etc, then go on to as you said what it's responsible for...)
- not necessarily limited to MRI (? already/soon)

MounaSafiHarab replied:
@davidblader 

maybe you can incorporate what @christinerogers mentioned above? 
I would personally not document the third item (it should be updated in the pull request that adds more than MRI support).

cmadjar replied:
Personally, I would replace the first sentence by:
LORIS-MRI is responsible for the insertion, visualization and archival of uploaded imaging datasets.

Two reasons:
1) no processing happens in LORIS, just conversion and insertion. Processing happens in CBRAIN or using other tools.
2) modifying "MRI scans" to "Imaging dataset" here as soon, we will support more than just MRI datasets.

christinerogers commented on docs/01-Introduction.md:
```diff
@@ -1,5 +1,55 @@
+
+### 1.2 How does LORIS-MRI work?
+![user_story](images/user_story.png)
+First, users upload a compressed (.tgz, .tar.gz, or .zip) DICOM folder to Imaging Uploader.
```
The back-end workflow... is there space to document here?  Or clarify that there is another route, to be discussed elsewhere.

davidblader replied:
mouna discusses that here: https://github.com/aces/Loris-MRI/pull/214

MounaSafiHarab replied:
maybe just add a note saying the `upload` process and its variations is discussed in more details in another section (you cna point to it).

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,5 +1,55 @@
+Assuming the upload went through successfully, an entry is added to the menu of Imaging Uploader
+and the upload's status throughout the pipeline can then be tracked by
+the user through the Log Viewer.
+
```
Deidentification of imaging datasets; please add a section on it; the text can be copied from:
https://github.com/aces/Loris-MRI/pull/219/commits/eaf9979ed91d1d47d46e7764d60964a95c7102c8

MounaSafiHarab replied:
Here you only comment on the filename, but also the DICOM headres need to be de-identified (please check the commit in my earlier comment for what can be added).

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+The pipeline can start automatically if the autolaunch configuration is set, otherwise
+ manual intervention is required by a developer with backend access to the server
+ (information on the scripts needed to run the pipeline can be found
+ in the Scripts section of the documentation).
```
can you please add a hyperlink?

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+ in the Scripts section of the documentation).
+
+ First, the upload is prepped to be archived on the server. The study
+ participant's data included in the upload (**DICOM headers?**) is crosschecked and
```
(PatientName or PatientID headers are checked, and that the upload contains nothing but DICOM files).

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+ participant's data included in the upload (**DICOM headers?**) is crosschecked and
+ validated with the LORIS database. Given that all uploaded metadata is valid, the
+ DICOM upload is archived on the server. The archival can be viewed and downloaded
+ through LORIS' DICOM Archive module.
```
the download option is not yet available!

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+ file formats. At this stage, scan parameters such as echo time, repetition time, etc.,
+ are checked with the MRI protocol table in the LORIS database to determine the scan types.
+
+If at either the archival or processing stages of the pipeline there is a failure,
```
- can you please change processing to pre-processing?

- this is not very accurate (at least not in the more modern way of running the pipeline) as archivals (and in fact upload!) won't proceed now if there are candidate errors. @cmadjar any idea how this statement can be made more accurate?

cmadjar replied:
Proposition:
"Before running the DICOM archival and MINC insertion stages, the uploaded imaging dataset will be validated to make sure that the study has not been fully inserted yet, that there are indeed DICOM files in the uploaded compressed file and that the patient name stored in the DICOM files matches the one entered when uploading the imaging dataset. If the uploaded compressed imaging dataset has not be validated, then DICOM archival and MINC insertion stages will not be run on this dataset and a log message will be available in the Log Viewer of the Imaging Uploader module.
However, if all went well during the first validation of the uploaded compressed file, then the DICOM archival and MINC insertion stages will be run. During the MINC insertion stage, if the scan type of a MINC image could not be determined using the `mri_protocol` table, then the scan will be inserted into the MRI violations LORIS module where details of the faulty scan can be viewed. If at least one scan successfully passed through the pipeline, the Progress column in the Imaging Uploader is set to success and successfully inserted scans are viewable in the Imaging Browser module. Each scan is then available to be navigated in BrainBrowser in 3D or 4D space. More details on the capabilities of BrainBrowser can be found [here.][5]"

@MounaSafiHarab this is probably more exact. What do you think?

MounaSafiHarab commented:
@davidblader 
I commented a bit on the pull request, but one more comment is related to the image; is there a way to add a dashed line from the radiologist to the Violated scans? these are also viewable from there in Brainbrowser and radiologists sometimes are required to review those as well (it happened in CCNA recently and they specifically asked to see the MINC file even if it violated the protocol parameters because it was an important structural modality).

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+ DICOM upload is archived on the server. The archival can be viewed and downloaded
+ through LORIS' DICOM Archive module.
+
+ Following archival, the scans are copied into MINC and (optionally) NII
```
replace "copied" by "converted"

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
-## 1.2 How does LORIS work?
+LORIS-MRI is responsible for the processing, visualizing, and archiving
+of uploaded MRI scans. It expects an uploaded, compressed file containing
+several [DICOM][2] files, processes this into [MINC][3] and [NII][4]
```
replace "processes" by "converts"

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+these file formats can be helpful, but are not necessary for using or
+installing LORIS-MRI.
 
+LORIS-MRI allows you to easily organize and archive your MRI data and links it with 
```
replace "MRI data" by "imaging datasets"

cmadjar replied:
...and links **them** with...

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,55 @@
+the successfully processed scans are now viewable in Imaging Browser. Each scan is
+ then available to be navigated in BrainBrowser in 3D or 4D space. More details
+ on the capabilities of BrainBrowser can be found [here.][5]
+
```
Regarding the picture attached, I find it difficult to read the content of the image. Is there a way to improve this?
Would also be nice to try to reduce the text in the figure if possible.

MounaSafiHarab commented:
@davidblader 

do you think you can take a look at the comments and incorporate them please? it would be good to have all the documentation merged so we can start refining, and do a first pass of proof-reading.

davidblader commented:
@MounaSafiHarab @cmadjar i think i've addressed your points, let me know if i missed something :smiley_cat: 
sorry that it took me forever to get to this

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,82 @@
+upload a compressed (.tgz, .tar.gz, or .zip) DICOM folder to Imaging
+Uploader.  LORIS requires that the uploaded file name follow the naming
+convention PSCID_CANDID_VISIT-LABEL.  Assuming the upload went through
+successfully, an entry is added to the menu of Imaging Uploader and the
```
...of *the* Imaging Uploader *module. The status of the upload during the insertion pipeline* can then be tracked by the user through the Log Viewer.

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,82 @@
+upload's status throughout the pipeline can then be tracked by the user
+through the Log Viewer.
+
+All DICOM datasets transferred to the DCC must be free of any identifying
```
Replace the "transferred to the DCC" by "uploaded via the Imaging Uploader or transferred on the LORIS-MRI server"

cmadjar replied:
No everyone uses the term DCC.

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,82 @@
+ Following archival, the scans are converted into MINC and (optionally)
+ NII file formats. At this stage, scan parameters such as echo time,
+ repetition time, etc., are checked with the MRI protocol table in the
+ LORIS database to determine the scan types.
```
Add the following at the end of this paragraph:
"Once the scan type has been determined, the scan are inserted into the database and viewable via the Imaging Browser module. If the scan type could not be determined, then the scan will be accessible via the MRI Violated Scans module."

cmadjar commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,82 @@
+ controlled through the user's web browser, facilitating collaboration
+ between radiologists, clinicians and researchers.
+
+### 1.2 How does LORIS-MRI work?  ![user_story](images/user_story.png)
```
Comments on the figure:
- I would remove the little guys. It takes a lot of space and does not add to the graph. Additionally, it is rare that the radiologist would do the QC of the images. They only check for incidental findings. In several studies, it would be someone else that does the QC of the images.
- in the attached screenshot are propositions for text edition
![user_story](https://user-images.githubusercontent.com/1402456/38314780-0c2ed31c-37f5-11e8-80ab-5c59b8f24360.png)

davidblader commented:
having trouble locating where i put that diagram file but i will definitely make the requested changes to it once i do

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,89 @@
+organization, and archival of uploaded imaging datasets.
+ It expects an uploaded, compressed file containing
+a [DICOM][2] study composed of several DICOM files. These DICOM files
+ will be archived on the server and converted to [MINC][3] and [NIfTI][4]
```
@davidblader 
please add the word optionally to NIfTI (it is optional depending on project config settings)

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,89 @@
+
+### 1.2 How does LORIS-MRI work?  ![user_story](images/user_story.png)
+LORIS-MRI allows multiple ways to upload scans, more details about these
+options can be found [here.](05-PipelineLaunchOptions.md) Typically, users
```
I would remove the sentence about 'more details about these options can be found here' and just move it to 2 paragraphs later (I will add a comment where I think it makes more sense to put it).
I would just say here:
"LORIS-MRI allows multiple ways to upload scans. But typically, users uploads etc,..."

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,89 @@
+The pipeline can start automatically if the autolaunch configuration is
+set, otherwise
+ manual intervention is required by a developer with backend access to
+ the server (information on the scripts needed to run the pipeline can
```
Here, I would tend to add the link to the section about how to launch the pipeline (the link you have is useful, but too general as it related to all the scripts, and some **re**insertion scripts, not just how to launch the pipeline)...

MounaSafiHarab commented on docs/01-Introduction.md:
```diff
@@ -1,6 +1,89 @@
+ the server (information on the scripts needed to run the pipeline can
+ be found in the [Scripts section](04-Scripts.md) of the documentation).
+
+#### General outline of the pipeline:
```
I feel this header is out of place; I somehow feel all the info below it can be summarized with 2 lines about imaging browser versus violated scans and added just before.

MounaSafiHarab commented:
can we change the base branch to be minor?

davidblader commented:
@cmadjar @MounaSafiHarab could you take another look?
